mulog-adv-console/src/com/brunobonacci/mulog/publishers/advanced_console.clj

```diff
@@ -0,0 +1,123 @@
+ (ns com.brunobonacci.mulog.publishers.advanced-console
+   (:require [com.brunobonacci.mulog.flakes :refer [flake?]]
+             [com.brunobonacci.mulog.publisher :as p]
+             [com.brunobonacci.mulog.buffer :as rb]
+             [clansi :refer [style]]))
+
+
+(defn wrap-quotes
+  [^String s]
+  (str "\"" s "\""))
+
+(defn colorize
+  [thing color]
+  (apply style (if (or (string? thing)
+                      (flake? thing))
+                (wrap-quotes thing)
+                thing)
+              (if (keyword? color)
+                [color]
+                color)))
+
+(defn colorize-item
+  [item color]
+  (reduce-kv (fn [acc k v]
+               (assoc acc
+                      (colorize k color)
+                      (colorize v color)))
+             {}
+             item))
+
+(defn naive-prettify
+  [items]
+  (as-> items $
+    (reduce-kv (fn [acc k v]
+                 (conj acc (str k " " v ",\n")))
+               ["{\n"]
+               $)
+    (conj $ "}")
+    (apply str $)))
+
+(def formatters
+  (atom {}))
```

Comment:
I think we should not have an atom for the formatters. This should be specified in the publisher.

Although it doesn't make much sene, however someone could start two ansi publishers with two different configuration. Having a single atom for all the formatters forces the could cause clashes.

I would rather move the formatters inside the config specified by the user.